server/data/statementsClient.ts

```diff
@@ -41,61 +41,74 @@ export default class StatementsClient {
-    from report r
-    left join statement s on r.id = s.report_id
-    where r.id = $1
-      and r.deleted is null
-      and s.user_id = $2
-      and s.statement_status = $3
-      and s.deleted is null`,
+            ,      r.booking_id             "bookingId"
+            ,      r.incident_date          "incidentDate"
+            ,      s.last_training_month    "lastTrainingMonth"
+            ,      s.last_training_year     "lastTrainingYear"
+            ,      s.job_start_year         "jobStartYear"
+            ,      s.statement
+            ,      s.submitted_date         "submittedDate"
+            ,      s.name                   "name"
+            ,      r.reporter_name          "reporterName"
+            from report r
+            left join statement s on r.id = s.report_id
+            where r.id = $1
+              and r.deleted is null
+              and s.user_id = $2
+              and s.statement_status = $3
+              and s.deleted is null`,
       values: [reportId, userId, status.value],
     })
     return results.rows[0]
   }
 
-  async getStatementForReviewer(statementId: number) {
+  async getStatementForReviewer(statementId: number): Promise<ReviewerStatement> {
     const results = await this.query({
       text: `select s.id
-    ,      r.id                     "reportId"
-    ,      s.name
-    ,      r.booking_id             "bookingId"
-    ,      r.incident_date          "incidentDate"
-    ,      s.last_training_month    "lastTrainingMonth"
-    ,      s.last_training_year     "lastTrainingYear"
-    ,      s.job_start_year         "jobStartYear"
-    ,      s.statement
-    ,      s.submitted_date         "submittedDate"
-    from report r
-    left join statement s on r.id = s.report_id
-    where s.id = $1
-    and s.deleted is null`,
-      values: [statementId],
+            ,      r.id                       "reportId"
+            ,      s.name
+            ,      s.user_id                  "userId"
+            ,      s.overdue_date <= now()    "isOverdue"
+            ,      s.statement_status = $1    "isSubmitted"
+            ,      s.statement_status = $2    "isDisputed"
+            ,      r.booking_id               "bookingId"
+            ,      r.incident_date            "incidentDate"
+            ,      s.last_training_month      "lastTrainingMonth"
+            ,      s.last_training_year       "lastTrainingYear"
+            ,      s.job_start_year           "jobStartYear"
+            ,      s.statement
+            ,      s.submitted_date           "submittedDate"
+            from v_report r
+            left join v_statement s on r.id = s.report_id
+            where s.id = $3`,
+      values: [StatementStatus.SUBMITTED.value, StatementStatus.DISPUTED.value, statementId],
     })
     return results.rows[0]
   }
 
-  async getStatementsForReviewer(reportId: number) {
+  async getStatementsForReviewer(reportId: number): Promise<ReviewerStatement[]> {
```

Comment:
I've aligned the return types of getStatementForReviewer and getStatementsForReviewer to reuse same type definitions

Reply:
cool